lib/modules/manager/terragrunt/modules.ts

```diff
@@ -68,18 +68,28 @@ export function analyseTerragruntModule(
     dep.currentValue = githubRefMatch.groups.tag;
     dep.datasource = GithubTagsDatasource.id;
   } else if (gitTagsRefMatch?.groups) {
-    dep.depType = 'gitTags';
-    if (gitTagsRefMatch.groups.path.includes('//')) {
+    const { url, host, path, tag } = gitTagsRefMatch.groups;
+    const containsSubDirectory = path.includes('//');
+    if (containsSubDirectory) {
       logger.debug('Terragrunt module contains subdirectory');
-      dep.depName = gitTagsRefMatch.groups.path.split('//')[0];
-      const tempLookupName = gitTagsRefMatch.groups.url.split('//');
-      dep.packageName = tempLookupName[0] + '//' + tempLookupName[1];
+    }
+    dep.depType = 'gitTags';
+    // We don't want to have .git or subdirectory in the depName
+    dep.depName = host + '/' + path.split('//')[0].replace('.git', '');
+    dep.currentValue = tag;
+    dep.datasource = detectGitTagDatasource(url);
+    if (dep.datasource === GitTagsDatasource.id) {
+      if (containsSubDirectory) {
+        const [protocol, hostAndPath] = url.split('//');
+        dep.packageName = protocol + '//' + hostAndPath;
+      } else {
+        dep.packageName = url;
+      }
     } else {
-      dep.depName = gitTagsRefMatch.groups.path.replace('.git', '');
-      dep.packageName = gitTagsRefMatch.groups.url;
+      // The packageName should only contain the path to the repository
+      dep.packageName = path.split('//')[0];
+      dep.registryUrls = [...(dep.registryUrls ?? []), `https://${host}`];
```

Comment:
are you sure it can already have registry urls? I don't think so 
```suggestion
      dep.registryUrls = [`https://${host}`];
```

Reply:
Just implemented to not break, just in case there is already a registryUrl present. I can also just change it to a simple assignment.

Reply:
where should it come from?